client/asset/eth/eth.go

```diff
@@ -2689,17 +2794,35 @@ func (w *assetWallet) SwapConfirmations(ctx context.Context, coinID dex.Bytes, c
 		return 0, false, fmt.Errorf("error fetching best header: %w", err)
 	}
 
-	swapData, err := w.swap(w.ctx, secretHash, contractVer)
+	status, err := w.status(w.ctx, secretHash, contractVer)
 	if err != nil {
 		return 0, false, fmt.Errorf("error finding swap state: %w", err)
 	}
 
-	if swapData.State == dexeth.SSNone {
+	if status.Step == dexeth.SSNone {
 		return 0, false, asset.ErrSwapNotInitiated
 	}
 
-	spent = swapData.State >= dexeth.SSRedeemed
-	confs = uint32(hdr.Number.Uint64() - swapData.BlockHeight + 1)
+	spent = status.Step >= dexeth.SSRedeemed
+	confs = uint32(hdr.Number.Uint64() - status.BlockHeight + 1)
+
+	// NOTE: confs will equal to the block number for the version 1 contract if spent == true,
+	// because BlockHeight will be zero. This is probably fine, since the caller
+	// will examine spent first? Otherwise, we could do this.
```

Comment:
Interesting problem to have here, but I think it's right to be using what's recorded in the contract state like you're doing in this version of the PR, at least before it's spent.  What you suggest about getting the tx confirmations after it is spent might be alright unless the tx was replaced.